azkaban-webserver/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -1747,12 +1747,12 @@ private void ajaxHandleUpload(HttpServletRequest req,
         }
         ret.put("error", "Installation Failed.<br>" + error);
       } finally {
-        if (tempDir.exists()) {
-          FileUtils.deleteDirectory(tempDir);
-        }
         if (out != null) {
           out.close();
         }
+        if (tempDir.exists()) {
+          FileUtils.deleteDirectory(tempDir);
```

Comment:
Perform out.close() before attempting FileUtils.deleteDirectory to avoid leaked file handle when deleteDirectory throws IOException.